### PR TITLE
add chaosmonkey type

### PR DESCRIPTION
**Pull request by glitchcrab**
I _think_ I've fudged my way through this correctly.

### CODE REVIEW

File: pkg/apis/chaosmonkey/v1alpha1/chaosmonkey_types.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package v1alpha1
```
Comment (rossf7) [inline pkg/apis/chaosmonkey/v1alpha1/chaosmonkey_types.go]: I would put this in example group. So the filename should be `pkg/apis/example/v1alpha1/chaosmonkey_types.go`

**Review by rossf7 — COMMENTED** [review]
@glitchcrab Direction is great but a few changes needed.

**Comment by glitchcrab** [comment]
well that was painful

File: pkg/apis/chaosmonkey/v1alpha1/chaosmonkey_types.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+type ChaosMonkeyConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              ChaosMonkeyConfigSpec `json:"spec"`
+	Status            ChaosMonkeyStatus     `json:"status"`
+}
+
+type ChaosMonkeyConfigSpec struct {
+	// StartTime is the beginning of the period when pods can be killed.
+	StartTime int `json:"starttime" yaml:"starttime"`
```
[the remark this answers is not in the document]
Comment (glitchcrab) [reply pkg/apis/chaosmonkey/v1alpha1/chaosmonkey_types.go]: @rossf7 can you suggest what format I should be using in my CR to satisfy DeepCopyTime? My intention with this field was for it to be an hour, as in allow the operator to delete pods between say 1000 and 1600.

[the remark this answers is not in the document]
Comment (rossf7) [reply pkg/apis/chaosmonkey/v1alpha1/chaosmonkey_types.go]: I think `1998-05-05T05:05:05Z` should work.

See https://github.com/kubernetes/apimachinery/blob/0ee8b4573e3aecbab8ba0d539f919aa7c53365b2/pkg/apis/meta/v1/time_test.go#L80

[the remark this answers is not in the document]
Comment (rossf7) [reply pkg/apis/chaosmonkey/v1alpha1/chaosmonkey_types.go]: If it's a time period then having the hours as string fields might work better.

**Comment by glitchcrab** [comment]
not required anymore

**Pull request closed without merging**